Using TCK Tested JDK builds of OpenJDK

carldea opened the pull request:
The AdoptOpenJDK has been discontinued since July 2021. When using Zulu you get all the latest updated (TCK Tested) builds for all versions of OpenJDK. Following best practices by adding fixed versions for LTS JDKs.

hazendaz commented:
@carldea looking at https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Adopt, the natural replacement is the continued product as 'temurin'.  Looking further to see zulu vs temurin, https://www.azul.com/products/core/jdk-comparison-matrix/.  So really in this use case either would be fine for what we are doing so no issues in using zulu.  However, don't think we want multiple jdk 8/11 runtimes shown here and no reason to run older copies.  The ones listed are rather old.  I would need to look a bit further but thought this already ran latest when just indicating the version as 8 for example.  If that is the case, I think we stay at that.  If we need to be more specific, then we should fully define the latest and presumably dependabot will auto update us going forwards.  Thoughts?

carldea commented:
> @carldea looking at https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Adopt, the natural replacement is the continued product as 'temurin'. Looking further to see zulu vs temurin, https://www.azul.com/products/core/jdk-comparison-matrix/. So really in this use case either would be fine for what we are doing so no issues in using zulu. However, don't think we want multiple jdk 8/11 runtimes shown here and no reason to run older copies. The ones listed are rather old. I would need to look a bit further but thought this already ran latest when just indicating the version as 8 for example. If that is the case, I think we stay at that. If we need to be more specific, then we should fully define the latest and presumably dependabot will auto update us going forwards. Thoughts?

Sounds like a plan! 
Commits Completed (Workflow runs all green).  I've removed the fixed versions, updated the comments and squashed commits. 👍🏼 

**My reasoning for adding a fixed versions for LTS versions:**
The reason I added them was often as a best practice when library owners have customers that run on a mature (fixed) version and getting the latest (super up to date build) can (and has) broken stuff in production.

hazendaz commented:
@carldea Thanks!
